sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -65,6 +65,21 @@ class FileIndexSuite extends SharedSQLContext {
     }
   }
 
+  test("SPARK-26990: use user specified field names if possible") {
+    withTempDir { dir =>
+      val partitionDirectory = new File(dir, "a=foo")
+      partitionDirectory.mkdir()
+      val file = new File(partitionDirectory, "text.txt")
+      stringToFile(file, "text")
+      val path = new Path(dir.getCanonicalPath)
+      val schema = StructType(Seq(StructField("A", StringType, false)))
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
```

Comment:
Also test the behavior when the conf is on.

Reply:
When the conf is on, the schema will be `a` instead of `A`. Both 2.4 and the current code has the same result.
Not sure if we should throw exceptions on this.